### ISSUE TITLE
Vlite issue

### ISSUE THREAD

**Issue by ZhiShengYuan** [v2fly/v2ray-core#2594]
Vlite protocol seems not work

tl;dr the server will panic when the client connect,and report ` handshake error: dtls temporary: packet length and declared length do not match panic: runtime error: invalid memory address or nil pointer dereference`

### To Reproduce:

Step 1: install v2ray 5.7.0 in client and server
Step 2: use this config 
server:
```json
{
  "inbounds": [
    {
      "protocol": "vliteu",
      "port":443,
      "listen":"0.0.0.0",
      "settings": {
        "password":"1145141919870",
        "scramblePacket":true,
        "enableStabilization":false,
        "enableRenegotiation":true,
        "enableFec":false,
        "handshakeMaskingPaddingSize":1280
      }
    }
  ],
  "outbounds": [
    {
      "protocol": "freedom",
      "tag": "direct"
  }
  ]
}
```
client:
```json
{
  "inbounds": [
  ],
  "outbounds": [
    {
      "protocol": "vliteu",
      "settings": {
        "address": "114514",
        "port": 443,
        "password": "1145141919810",
        "scramblePacket": true,
        "enableFec": false,
        "enableStabilization": true,
        "enableRenegotiation": false,
        "handshakeMaskingPaddingSize": 1280
      },
      "tag": "out_vlite"
    }
  ],
  "services": {}
}
```
Step 3: start server and watch,once the client connect,the server will exit and output some error info like this 
```
V2Ray 5.7.0 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.20.4 linux/amd64)
A unified platform for anti-censorship.
2023/07/08 11:24:39 [Warning] V2Ray 5.7.0 started
2023/07/08 11:24:39 handshake error: dtls temporary: packet length and declared length do not match
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xceef4e]

goroutine 21 [running]:
github.com/pion/dtls/v2.(*Conn).isHandshakeCompletedSuccessfully(...)
        github.com/pion/dtls/v2@v2.2.4/conn.go:815
github.com/pion/dtls/v2.(*Conn).Read(0x1b91f78?, {0xc0004de000?, 0x44aeb4?, 0x5d9008?})
        github.com/pion/dtls/v2@v2.2.4/conn.go:293 +0x2e
github.com/xiaokangwang/VLite/transport/filteredConn.(*FilteredConn).Read(0xc0003ecaa0, {0xc00024c000, 0x2000, 0xc0004729c0?})
        github.com/xiaokangwang/VLite@v0.0.0-20220418190619-cff95160a432/transport/filteredConn/conn.go:147 +0x7c
github.com/pion/sctp.(*Association).readLoop(0xc0003d4680)
        github.com/pion/sctp@v1.7.6/association.go:421 +0x14e
created by github.com/pion/sctp.(*Association).init
        github.com/pion/sctp@v1.7.6/association.go:298 +0xd0
```
...

### Application Info
Debian 12  `Linux CSTNET 6.1.0-9-cloud-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.27-1 (2023-05-08) x86_64 GNU/Linux` , V2Ray 5.7.0 (V2Fly, a community-driven edition of V2Ray.) Custom (go1.20.4 linux/amd64)

**Comment by Go-to-Gitea-already** [comment]
github.com/pion/dtls/v2.(*Conn).isHandshakeCompletedSuccessfully(...)
	github.com/pion/dtls/v2@v2.2.4/conn.go:815

**Comment by Go-to-Gitea-already** [comment]
UPD: was a mistake. vlite falling only with TCP traffic. What about UDP, i dont know, because i dont have installed wireshark and v2ray cannot sniff UPD :-)
So i will install it and maybe open new issuse.
